partcad/src/partcad/cache_shape.py

```
@@ -0,0 +1,130 @@
+                results[key] = None
+                in_memory[key] = False
+                continue
+
+            # TODO(clairbee): if we know this is a valid solid, then we should use faster BREP
+            # if key.startswith(["part", "assembly"]):
+            #     # serialization = SERIALIZATION_BREP
+            # else:
+            serialization = self.serialization
+            if serialization == SERIALIZATION_PICKLE:
+                try:
+                    obj = pickle.loads(data)
```

Ruff (0.8.2)

`pickle` and modules that wrap it can be unsafe when used to deserialize untrusted data, possible security issue

(S301)